worldengine/draw.py

```diff
@@ -161,14 +163,47 @@ def __setitem__(self, item, value):
         self.pixels[item] = value
 
 
-def draw_simple_elevation(data, width, height, sea_level, target):
+def draw_simple_elevation(world, sea_level, target):
     """ This function can be used on a generic canvas (either an image to save
         on disk or a canvas part of a GUI)
     """
-    for y in range(height):
-        for x in range(width):
-            e = data[y][x]#TODO: numpy-notation [y,x] is a little faster; all passed arrays would have to be numpy-arrays
-            r, g, b = elevation_color(e, sea_level)
+    min_elev_sea = None
+    max_elev_sea = None
+    min_elev_land = None
+    max_elev_land = None
+    for y in range(world.height):
+        for x in range(world.width):
+            e = world.elevation['data'][y,x]
+            if sea_level is None:
+                if min_elev_land is None or e < min_elev_land:
+                    min_elev_land = e
+                if max_elev_land is None or e > max_elev_land:
+                    max_elev_land = e
+            elif world.is_land((x, y)):
+                if min_elev_land is None or e < min_elev_land:
+                    min_elev_land = e
+                if max_elev_land is None or e > max_elev_land:
+                    max_elev_land = e
+            else:
+                if min_elev_sea is None or e < min_elev_sea:
+                    min_elev_sea = e
+                if max_elev_sea is None or e > max_elev_sea:
+                    max_elev_sea = e
+
+    elev_delta_land = (max_elev_land - min_elev_land)/11
+    if sea_level != None:
+        elev_delta_sea = max_elev_sea - min_elev_sea
+    
+    for y in range(world.height):
```

Comment:
Why don't I leave that to you, since you have more experience than I do with numpy?

This process has given me much better insight into the commands for coverage testing and I'm now able to test coverage locally. As a result I would like to spend a little time creating some new tests to expand coverage. Biome's only testing about 70% right now.